recipes/libgphoto2/all/conanfile.py

```diff
@@ -13,122 +19,113 @@ class LibGphoto2(ConanFile):
-    settings = "os", "compiler", "build_type", "arch"
+    package_type = "shared-library"
+    settings = "os", "arch", "compiler", "build_type"
     options = {
-        "shared": [True, False],
-        "fPIC": [True, False],
         "with_libusb": [True, False],
         "with_libcurl": [True, False],
         "with_libxml2": [True, False],
         "with_libexif": [True, False],
         "with_libjpeg": [True, False],
     }
     default_options = {
-        "shared": True,
-        "fPIC": True,
         "with_libusb": True,
         "with_libcurl": True,
         "with_libxml2": True,
         "with_libexif": True,
         "with_libjpeg": True,
     }
-    generators = "pkg_config"
-    _autotools = None
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
 
-    def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-
-    def validate(self):
-        if not self.options.shared:
-            raise ConanInvalidConfiguration("building libgphoto2 as a static library is not supported")
-        if self.settings.os == "Windows":
-            raise ConanInvalidConfiguration("libgphoto2 does not support Windows")
-
-    @contextlib.contextmanager
-    def _build_context(self):
-        if self.settings.os == "Linux" and not self.options["libtool"].shared:
-            env = {
-                "LIBLTDL": "-lltdl -ldl",
-            }
-            with tools.environment_append(env):
-                yield
-        else:
-            yield
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def configure(self):
-        if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
 
-    def build_requirements(self):
-        self.build_requires("pkgconf/1.7.4")
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("libtool/2.4.6")
+        self.requires("libtool/2.4.7")
         if self.options.with_libusb:
             self.requires("libusb/1.0.26")
         if self.options.with_libcurl:
-            self.requires("libcurl/7.83.1")
+            self.requires("libcurl/8.0.0")
         if self.options.with_libxml2:
-            self.requires("libxml2/2.9.14")
+            self.requires("libxml2/2.10.3")
         if self.options.with_libexif:
             self.requires("libexif/0.6.23")
         if self.options.with_libjpeg:
-            self.requires("libjpeg/9d")
+            self.requires("libjpeg/9e")
+
+    def validate(self):
+        if is_msvc(self):
+            raise ConanInvalidConfiguration("Visual Studio not supported yet")
```

Comment:
Lets encourage them to contribute if they see this message :)

```suggestion
            raise ConanInvalidConfiguration(f"{self.ref} currently Visual Studio is not supported. Contributions are welcomed.")
```